cxx/distributions/zero_mean_normal.hh

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#pragma once
+#include <random>
+#include <tuple>
+#include <variant>
+
+#include "base.hh"
+#include "util_math.hh"
+
+// A normal distribution that is known to have zero mean.
+// This class is not intended to be directly used as a data model, but
+// rather to support the Gaussian emissions model.
+class ZeroMeanNormal : public Distribution<double> {
```

Comment:
Is it possible to add an arg to Normal to update the mean / not update the mean (and that way we can avoid adding tests / duplicating code)?

Reply:
That would work for incorporate / unincorporate, but I don't think it would work for logp, logp_score or sample.  You get different posterior predictive and marginal likelihood distributions when you have to integrate out the mean versus when you don't.